led_strip/include/led_strip_spi.h

```diff
@@ -0,0 +1,52 @@
+/*
+ * SPDX-FileCopyrightText: 2022-2023 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#pragma once
+
+#include <stdint.h>
+#include "esp_err.h"
+#include "driver/gpio.h"
+#include "esp_rom_gpio.h"
+#include "driver/spi_master.h"
+#include "driver/spi_common.h"
+#include "hal/spi_types.h"
+#include "led_strip_types.h"
+#include "soc/gpio_sig_map.h"
+#include "soc/spi_periph.h"
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief LED Strip SPI specific configuration
+ */
+typedef struct {
+    rmt_clock_source_t clk_src; /*!< SPI clock source */
+    uint32_t resolution_hz;     /*!< SPI tick resolution, if set to zero, a default resolution (10MHz) will be applied */
+    size_t mem_block_symbols;   /*!< How many RMT symbols can one RMT channel hold at one time. Set to 0 will fallback to use the default size. */
+    spi_host_device_t host_channel ;
```

Comment:
unnecessary space before the ";". also suggest not using the word "channel". SPI doesn't have the concept of "channel". Here you may want to use "host controller"